### ISSUE TITLE
Allow partial application / currying

### ISSUE THREAD

**Issue by Planeshifter** [compute-io/polynomial#2]
Would be great if one could simply call `polyval( coef  )` and get a new function back with the specified coefficients. Although one can do this simply by using the `.bind()` method, I believe this would be more convenient and natural.

**Comment by kgryte** [comment]
Yes, this seems reasonable. I would probably opt for an exported `factory` method (for inspiration, see [dstructs-ndarray](https://github.com/dstructs/ndarray#ndarray-factory)). I would like to avoid `.bind`, as this is a perf killer.
